cxx/gendb.cc

```diff
@@ -40,7 +40,8 @@ double GenDB::logp_score() const {
 
 void GenDB::incorporate(
     std::mt19937* prng,
-    const std::pair<int, std::map<std::string, ObservationVariant>>& row) {
+    const std::pair<int, std::map<std::string, ObservationVariant>>& row,
+    bool sample_new) {
```

Comment:
The name `sample_new` is confusing to me, since we're guaranteed to get "new" unseen values when it's false, and we might get previously-seen values when it's true. Could you use maybe `use_sequential_entities` or `use_unique_entities` (with the semantics flipped), or another name that's more explanatory?

Reply:
How do you like new_entities_have_new_parts?  (It has reversed semantics.)

Reply:
I don't understand what "parts" means or what a "new entity" is (is it an entity corresponding to a newly observed row or is it an entity that hasn't been seen before)? I think `use_unique_entities` or `initialize_with_unique_entities` does the job (or use your proposal, with a clarifying comment)

Reply:
Just saw the comment re. "parts" -- I think we should emphasize somewhere that new entities are also used for new rows (and it's still unclear to me what it means for an entity to have a part, which I understand as an attribute, so I'd lean towards one of the other names I proposed with "unique").

Reply:
How about new_rows_have_unique_entities?

Reply:
SGTM (though please add a comment that two entities that are added to the same domain in the course of adding a single row are also unique -- this is why I prefer a more general name like `use_unique_entities`).

Reply:
Done.